pkg/reconcile/pipeline/context/application.go

```diff
@@ -24,6 +20,7 @@ type application struct {
 	resource               *unstructured.Unstructured
 	bindingPath            *v1alpha1.BindingPath
 	bindableContainerNames sets.String
+	resourceMapping        pipeline.WorkloadMapping
```

Comment:
I made `pipeline.WorkloadMapping` objects represent a parsed, valid form of a `ClusterWorkloadResourceMappingTemplate` resource.  This gives me the following:
- I don't have to re-validate my `ClusterWorkloadResourceMappingTemplate`; if I have a `pipeline.WorkloadMapping` object, I know it's valid.
- Parsing `ClusterWorkloadResourceMappingTemplate` objects isn't free, since we need to parse our jsonpaths into Go data structures and the restricted jsonpaths into `[]string`.  I'd rather not waste computation time re-parsing these when I only have to do that once.